auth0/resource_auth0_organization.go

```diff
@@ -210,7 +153,53 @@ func deleteOrganization(ctx context.Context, d *schema.ResourceData, m interface
 	return nil
 }
 
-func buildOrganization(d *schema.ResourceData) *management.Organization {
+func updateOrganizationConnections(d *schema.ResourceData, api *management.Management) error {
+	toAdd, toRemove := Diff(d, "connections")
+
+	var err error
+	toRemove.Elem(func(data ResourceData) {
+		oldConnectionID, _ := data.GetChange("connection_id")
+
+		err = api.Organization.DeleteConnection(d.Id(), oldConnectionID.(string))
+		if err != nil {
+			return
+		}
+	})
+	if err != nil {
+		return err
+	}
+
+	toAdd.Elem(func(data ResourceData) {
+		organizationConnection := &management.OrganizationConnection{
+			ConnectionID:            String(data, "connection_id"),
+			AssignMembershipOnLogin: Bool(data, "assign_membership_on_login"),
+		}
+
+		err = api.Organization.AddConnection(d.Id(), organizationConnection)
+		if err != nil {
+			return
+		}
+	})
+	if err != nil {
+		return err
+	}
+
+	Set(d, "connections").Elem(func(data ResourceData) {
```

Comment:
It was also important to remove the HasChange() from here, so that now we always loop through the final connections and send a PATCH with whatever the value is, as this operation is idempotent. This ultimately makes us correctly update the values.